src/wp-includes/webfonts-api/class-wp-webfonts-registry.php

```diff
@@ -0,0 +1,253 @@
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $font_family Family font to fetch.
+	 * @return array[] Registered webfonts.
+	 */
+	public function get_by_font_family( $font_family ) {
+		if ( ! is_string( $font_family ) || '' === $font_family ) {
+			return array();
+		}
+
+		$font_family_key = $this->convert_font_family_into_key( $font_family );
+
+		// If the font family is not registered, bail out.
+		if ( ! isset( $this->registry_by_font_family[ $font_family_key ] ) ) {
+			return array();
+		}
+
+		$webfonts = array();
+		foreach ( $this->registry_by_font_family[ $font_family_key ] as $registration_key ) {
+			// Safeguard. Skip if not in registry.
+			if ( ! isset( $this->registered[ $registration_key ] ) ) {
+				continue;
+			}
+
+			$webfonts[ $registration_key ] = $this->registered[ $registration_key ];
+		}
+
+		return $webfonts;
+	}
+
+	/**
+	 * Registers the given webfont if its schema is valid.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array $webfont Webfont definition.
+	 * @return string Registration key.
+	 */
+	public function register( array $webfont ) {
+		$webfont = $this->convert_to_kebab_case( $webfont );
+
+		// Validate schema.
+		if ( ! $this->validator->is_valid_schema( $webfont ) ) {
+			return '';
+		}
+
+		$webfont = $this->validator->set_valid_properties( $webfont );
+
+		// Add to registry.
+		$registration_key = $this->generate_registration_key( $webfont );
+		if ( isset( $this->registered[ $registration_key ] ) ) {
+			return $registration_key;
+		}
+
+		$this->registered[ $registration_key ] = $webfont;
+		$this->store_in_query_by_containers( $webfont, $registration_key );
+
+		return $registration_key;
+	}
+
+	/**
+	 * Store the webfont into each query by container.
+	 *
+	 * These containers provide a performant way to quickly query webfonts by
+	 * provider or font-family. The registration keys are stored in each for
+	 * O(1) lookup.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array  $webfont          Webfont definition.
+	 * @param string $registration_key Webfont's registration key.
+	 */
+	private function store_in_query_by_containers( array $webfont, $registration_key ) {
+		$font_family = $this->convert_font_family_into_key( $webfont['font-family'] );
+		$provider    = $webfont['provider'];
+
+		// Initialize the arrays if they do not exist.
+		if ( ! isset( $this->registry_by_provider[ $provider ] ) ) {
+			$this->registry_by_provider[ $provider ] = array();
+		}
+		if ( ! isset( $this->registry_by_font_family[ $font_family ] ) ) {
+			$this->registry_by_font_family[ $font_family ] = array();
+		}
+
+		$this->registry_by_provider[ $provider ][]       = $registration_key;
+		$this->registry_by_font_family[ $font_family ][] = $registration_key;
+	}
+
+	/**
+	 * Convert camelCase parameters into kebab_case.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string[] $webfont Webfont definition.
+	 * @return array Webfont with kebab_case parameters (keys).
+	 */
+	private function convert_to_kebab_case( array $webfont ) {
```

Comment:
@aristath Reopening this here, since it doesn't look like this has been changed or received any feedback why it shouldn't be changed. Could you please follow-up here?

Reply:
@felixarntz  This is needed to prepare for the CSS as `fontFamily` is not a valid CSS property. Knowing that the `theme.json` will be camelCase, i.e. `fontFamily`, we thought to keep the registration process the same across `theme.json` and non-`theme.json` implementations. Why? To make it easier on theme authors when they support multiple versions of their themes.

The conversion here ensures the webfonts are ready for CSS processing, making it agnostic from where it's registered (i.e. `theme.json` or via an array elsewhere).

Given this extra bit of information, is okay to leave this conversion in place within the API to avoid pushing it upstream?

Reply:
@hellofromtonya I just realized now that the intention is to actually use camel-case names for all the arguments. I have a strong concern about that which I voiced in https://github.com/WordPress/wordpress-develop/pull/1736#discussion_r743095196 - this breaks the de-facto convention in WP core's PHP code to use snake-case.

If we use snake-case, then I definitely see the value of this method here, replacing from snake-case to hyphen-case. Let's move the discussion here to https://github.com/WordPress/wordpress-develop/pull/1736#discussion_r743095196, which is probably the bigger point in question.